a10_octavia/controller/worker/tasks/handler_virtual_server.py

```diff
@@ -12,91 +12,77 @@
+            LOG.debug("LoadBalancer created/updated succesfully: %s",
+                      loadbalancer.id)
         except Exception as e:
-            LOG.error(str(e))
-            status = {'loadbalancers': [{"id": loadbalancer_id,
-                                         "provisioning_status": constants.ERROR}]}
-        LOG.info(str(status))
-        return status
-
-    def revert(self, loadbalancer_id, *args, **kwargs):
-        pass
+            LOG.exception("Failed to create/update load balancer: %s", str(e))
+            raise
 
 
 class CreateVirtualServerTask(LoadBalancerParent, BaseVThunderTask):
+
     """ Task to create a virtual server """
 
-    def execute(self, loadbalancer_id, loadbalancer, vthunder):
+    def execute(self, loadbalancer, vthunder):
         c = self.client_factory(vthunder)
-        status = self.set(c.slb.virtual_server.create,
-                          loadbalancer_id,
-                          loadbalancer,
-                          vthunder)
-        return status
+        self.set(c.slb.virtual_server.create, loadbalancer, vthunder)
+
+    def revert(self, loadbalancer, vthunder, *args, **kwargs):
+        c = self.client_factory(vthunder)
+        try:
+            c.slb.virtual_server.delete(loadbalancer.id)
+        except Exception as e:
+            LOG.exception("Failed to revert create load balancer: %s", str(e))
 
 
 class DeleteVirtualServerTask(BaseVThunderTask):
+
     """ Task to delete a virtual server """
 
     def execute(self, loadbalancer, vthunder):
-        loadbalancer_id = loadbalancer.id
+        c = self.client_factory(vthunder)
         try:
-            c = self.client_factory(vthunder)
-            c.slb.virtual_server.delete(loadbalancer_id)
-            status = {'loadbalancers': [{"id": loadbalancer_id,
-                                         "provisioning_status": constants.DELETED}]}
+            c.slb.virtual_server.delete(loadbalancer.id)
         except Exception as e:
-            LOG.error(str(e))
-            status = {'loadbalancers': [{"id": loadbalancer_id,
-                                         "provisioning_status": constants.ERROR}]}
-        LOG.info(str(status))
-        return status
-
-    def revert(self, loadbalancer, *args, **kwargs):
-        pass
+            LOG.warning("Failed to delete load balancer: %s", str(e))
 
 
 class UpdateVirtualServerTask(LoadBalancerParent, BaseVThunderTask):
+
     """ Task to update a virtual server """
 
     def execute(self, loadbalancer, vthunder):
         c = self.client_factory(vthunder)
         status = self.set(c.slb.virtual_server.update,
-                          loadbalancer.id,
                           loadbalancer,
                           vthunder)
-        return status
 
-    def revert(self, loadbalancer, vthunder, update_dict):
-        pass
+    def revert(self, loadbalancer, vthunder, *args, **kwargs):
```

Comment:
We don't need to log this failure here. The failure will propagate up. Only need to revert in the case of orphaned resources.